BookWorm/BookWorm/View Controllers/SearchDetailsViewController.m

```diff
@@ -64,25 +100,48 @@ - (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger
 
 - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
     UITableViewCell *cell = [self.optionsTableView cellForRowAtIndexPath:indexPath];
+    
+    // getting the previous title of the button
+    NSString *prevTitle = self.markThisBookButton.titleLabel.text;
+    
+    // setting the new title of the button
     [self.markThisBookButton setTitle:cell.textLabel.text forState:UIControlStateNormal];
     self.optionsTableView.hidden = YES;
     
+    NSString *currTitle = [self.markThisBookButton currentTitle];
+    self.buttonString = currTitle;
     
-     PFUser *currUser = [PFUser currentUser];
+    currTitle = [currTitle stringByReplacingOccurrencesOfString:@" " withString:@""];
+    prevTitle = [prevTitle stringByReplacingOccurrencesOfString:@" " withString:@""];
+        
+    PFUser *currUser = [PFUser currentUser];
     
-    if ([[self.markThisBookButton currentTitle] isEqualToString:@"To Read"]) {
-        currUser[@"toRead"] = @YES;
-    } else if ([[self.markThisBookButton currentTitle] isEqualToString:@"Read"]) {
-        currUser[@"read"] = @YES;
-    } else if ([[self.markThisBookButton currentTitle] isEqualToString:@"Reading"]) {
-        currUser[@"reading"] = @YES;
+    // no change to be made
+    if ([currTitle isEqualToString:prevTitle]) {
+        return;
+    } else if ([currTitle isEqualToString:@"Markthisbook"]) {
+        // We are unmarking this book. This means we want to remove
+        // the GoogleBook pointer for the previous array and delete
+        // the book from the Google Book database
+        [currUser removeObject:self.bookPassed forKey:prevTitle];
+    } else if ([prevTitle isEqualToString:@"Markthisbook"]) {
+        // We just add the object to the list we want to add it too
+        [currUser addUniqueObject:self.bookPassed forKey:currTitle];
     } else {
-        currUser[@"read"] = @NO;
-        currUser[@"toRead"] = @NO;
-        currUser[@"reading"] = @NO;
+        // changing from one list to another. Remove the book from one list.
+        // Put the pointer to the same book in another list
+        [currUser removeObject:self.bookPassed forKey:prevTitle];
+        [currUser saveInBackground];
+        [currUser addUniqueObject:self.bookPassed forKey:currTitle];
     }
     
-    [currUser saveInBackground];
+    [currUser saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error) {
```

Comment:
use [currUser saveInBackground]; instead if we don't need to handle callback, such as updating UI

Reply:
In this case, I am going to be updating UI. So, I'm going to be keeping this method.